guide/creating_element/synchronization.md

```diff
@@ -0,0 +1,224 @@
+             latency
+           ]) do
+    # ...
+```
+
+From this point, when PortAudio consumes audio it sends the update message to this clock.
+
+Here's the snippet from the native part of PortAudio sink. It's a callback
+called by PortAudio when an output device demands more audio data. To reduce the amount of messages sent, the actual message is sent every 100th invocation of the callback.
+
+```c
+#define SAMPLES_PER_MSEC 48
+
+static int callback(const void *_input_buffer, void *output_buffer,
+                    unsigned long frames_per_buffer,
+                    const PaStreamCallbackTimeInfo *_time_info,
+                    PaStreamCallbackFlags _flags, void *user_data) {
+  // ...
+
+  SinkState *state = (SinkState *)user_data;
+
+  if (++state->ticks % 100 == 0) {
+    send_membrane_clock_update(env, state->clock, UNIFEX_SEND_THREADED,
+                               100 * frames_per_buffer, SAMPLES_PER_MSEC);
+  }
+
+  // ...
+}
+```
+
+The message format sent by `send_membrane_clock_update` is defined in `sink.spec.exs`.
+As you can see in the snippet below, the `:membrane_clock_update` message contains
+a tuple with a number of frames and number of samples per millisecond.
+We could divide frames by samples and get time in milliseconds,
+but instead we send both values (that can be interpreted as numerator and denominator) -
+this way the clock can ensure that division rounding error won't affect accuracy of the clock.
+
+```elixir
+sends {:membrane_clock_update :: label, {frames :: int, sample_rate_ms :: int}}
+```
+
+The clock process accepts updates in different representations of time to next tick:
+
+* a single integer with time in milliseconds
+* tuple with numerator and denominator (used above)
+* rational number created by `Ratio` library (`t:Ratio.t/0`) - it can keep simplified fraction (2 integers)
+  if needed to prevent rounding
+
+They are described by `t:Membrane.Clock.update_t/0` type.
+
+## Timers - using a clock
+
+Each element can use the clock provided by a pipeline by setting up the _timer_
+Timer is a process that sends ticks in intervals set when it is started.
+This can be done by returning `t:Membrane.Element.Action.start_timer_t/0` action that requires a tuple containing:
+
+* an atom - id for a new timer,
+* time interval between ticks in (that can be of type `t:Membrane.Time.t/0` or `t:Ratio.t/0` where the numerator is `t:Membrane.Time.t/0`)
+* (optionally) pid of a clock that should be used by the timer. If no clock provided, a pipeline clock is used.
+
+Starting a clock means that the element will start receiving ticks that should be handled by a new callback -
+`c:Membrane.Element.Base.handle_tick/3` receiving timer id along with context and state.
+
+The timer can be stopped by `t:Membrane.Element.Action.stop_timer_t/0` action with id of a timer.
+
+### Example
+
+Here's a simplified example of a sink element that needs to consume video frames in proper speed and uses timer for that.
```

Comment:
How about: `Here's a simplified example of a sink element that uses timer to consume video frames in proper speed.`